traveler_ui_kit/src/main/java/com/guestlogix/traveleruikit/fragments/ParkingItemDetailsFragment.java

```diff
@@ -0,0 +1,198 @@
+import com.guestlogix.travelercorekit.utilities.DateHelper;
+import com.guestlogix.traveleruikit.R;
+
+import java.util.Date;
+import java.util.List;
+
+public class ParkingItemDetailsFragment extends Fragment
+        implements OnMapReadyCallback {
+    private static final String TAG = "BookingItemDetailsFragm";
+    private static final String ARG_PARKING_ITEM = "ARG_PARKING_ITEM";
+    private static final String ARG_PARKING_ITEM_DETAILS = "ARG_PARKING_ITEM_DETAILS";
+    private static final int ZOOM_LEVEL = 15;
+
+    //TODO replace heavy map fragment with a static image
+    private ParkingItemDetails parkingItemDetails;
+    private LinearLayout infoLinearLayout;
+    private TextView seeMoreLessToggle;
+    private boolean isParkingExpanded = true;
+
+    public static ParkingItemDetailsFragment newInstance(ParkingItem item, CatalogItemDetails details) {
+        Bundle args = new Bundle();
+        args.putSerializable(ARG_PARKING_ITEM, item);
+        args.putSerializable(ARG_PARKING_ITEM_DETAILS, details);
+        ParkingItemDetailsFragment fragment = new ParkingItemDetailsFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_parking_item_details, container, false);
+
+        Bundle args = getArguments();
+
+        if (args == null || !args.containsKey(ARG_PARKING_ITEM_DETAILS)) {
+            Log.e(TAG, "No ParkingItemDetails in arguments");
+            return null;
+        }
+
+        if (!args.containsKey(ARG_PARKING_ITEM)) {
+            Log.e(TAG, "No Parking Item in arguments");
+            return null;
+        }
+
+        parkingItemDetails = (ParkingItemDetails) args.get(ARG_PARKING_ITEM_DETAILS);
+
+        if (parkingItemDetails == null) {
+            Log.e(TAG, "No CatalogItemDetails");
+            return view;
+        }
+
+        infoLinearLayout = view.findViewById(R.id.linearLayout_parking_details_information);
+
+        seeMoreLessToggle = view.findViewById(R.id.textView_parking_details_general_information_toggle);
+        seeMoreLessToggle.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if(isParkingExpanded){
+                    seeMoreLessToggle.setText(getText(R.string.show_more));
+                    infoLinearLayout.setVisibility(View.GONE);
+
+                } else {
+                    seeMoreLessToggle.setText(getText(R.string.show_less));
+                    infoLinearLayout.setVisibility(View.VISIBLE);
+                }
+            }
+        });
+
+        setParkingDetails(view);
+        return view;
+    }
+
+    private void setParkingDetails(View view) {
+        TextView titleTextView = view.findViewById(R.id.textView_parking_details_title);
```

Comment:
Whats going on here? Seems like a lot of specific logic shoved into one method...it isnt very clear whats going on here..reevaluate the pattern here

Reply:
separated each of the setFoobar methods.

Reply:
✓